src/factories/node.ts

```diff
@@ -62,7 +65,10 @@ export async function nodeContainerFactory(node: RunGraphNode) {
 
     cacheKey = currentCacheKey
 
-    await renderNode(node)
+    await Promise.all([
+      renderNode(node),
+      createArtifacts(node.artifacts),
```

Comment:
You'll need to add something about the artifacts to the cache key otherwise artifacts updating won't cause a render.